### PR TITLE
Resilience4j fallback support

### PR DESCRIPTION
**Pull request by AmandeepSingh97**
Adding Resilience4j fallback support:
- Fallback
- Fallback factory

### CODE REVIEW

**Comment by fabiocarvalho777** [comment]
@Man-Has-No-Name thanks!! Do you plan to push more changes to add support for Resilience4J fallback support to Mocca? If yes, then let's save this PR and merge it together with the other changes. Is that ok?

File: mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java [hunk 1]
```diff
@@ -126,5 +129,43 @@ public Builder bulkhead(final Bulkhead bulkhead) {
             this.feignBuilder = feignBuilder.withBulkhead(bulkhead);
             return this;
```
Comment (fabiocarvalho777) [inline mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java:131]: Please, add `Fallback` on line 44 above.
Also, please don't forget to change end user documentation markdown file.

Thanks.

**Review by fabiocarvalho777 — CHANGES_REQUESTED** [review]
I see this PR is still marked as draft.
Please let me know when it is ready for review.
Meanwhile I added a couple of small comments.

Thanks.

File: mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java [hunk 2]
```diff
@@ -126,5 +129,43 @@ public Builder bulkhead(final Bulkhead bulkhead) {
...
         }
+
+        /**
+         *
+         * @param fallBack the fallback object to be set to in this builder
+         * @return
+         */
+        public Builder fallback(final Object fallBack) {
```
Comment (abhijith-prabhakar) [inline mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java:140]: what does fallback mean in this context?  For all scenarios?

File: mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java [hunk 3]
```diff
@@ -126,5 +129,43 @@ public Builder bulkhead(final Bulkhead bulkhead) {
             this.feignBuilder = feignBuilder.withBulkhead(bulkhead);
             return this;
         }
+
+        /**
+         *
+         * @param fallBack the fallback object to be set to in this builder
```
Comment (fabiocarvalho777) [inline mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java:137]: Please, add proper javadoc. Everything must be set, method description, and all tags, including `return`.

Do the same for all public methods you added here.

**Review by fabiocarvalho777 — CHANGES_REQUESTED** [review]

File: mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java (hunk 2 above)
Comment (AmandeepSingh97) [reply mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java:140]: As this does not involve any filter for a particular exception, the fallback added using this API will be triggered for all exceptions.

File: mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java (hunk 3 above)
Comment (AmandeepSingh97) [reply mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java:137]: Added javadoc for all newly added APIs
Also updated the end user documentation

File: docs/END_USER_DOCUMENT.md [hunk 4]
```diff
@@ -612,6 +614,23 @@ MoccaResilience4j moccaResilience2 = new MoccaResilience4j.Builder()
     .build();
 ```
 
+#### 6.5.2 Adding fallback to your client
```
Comment (fabiocarvalho777) [inline docs/END_USER_DOCUMENT.md]: Please remove this section, it is not necessary.
Instead, just add `.fallback(fallback)` in the example under section 6.5.1.

Also, notice the paragraph below should be under section 6.5.1. You removed it (probably accidentally).

```
Notice the instantiation and configuration of `rateLimiter` and `circuitBreaker` were omitted for brevity.
```

File: mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java [hunk 5]
```diff
@@ -126,5 +130,78 @@ public Builder bulkhead(final Bulkhead bulkhead) {
...
+                                       Class<? extends Exception> filter) {
+            this.feignBuilder = feignBuilder.withFallbackFactory(fallbackFactory, filter);
+            return this;
+        }
+
+        /**
+         * Adds a Fallback factory to the decorator chain using the exception filter predicate
+         *
+         * @param fallbackFactory the fallback factory function to be set in this builder
+         * @param filter Predicate for the exception filter
```
Comment (fabiocarvalho777) [inline mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java]: `predicate` in lower case please. Always start the parameter javadoc in lower case.

Please do the same in the other methods you added.

Thanks.

**Review by fabiocarvalho777 — CHANGES_REQUESTED** [review]

File: mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java (hunk 3 above)
Comment (fabiocarvalho777) [reply mocca-resiliency4j/src/main/java/com/paypal/mocca/client/MoccaResilience4j.java:137]: Thank you

**Review by fabiocarvalho777 — APPROVED** [review]

**Comment by fabiocarvalho777** [comment]
@abhijith-prabhakar I've approved this PR. Just pending your review now to have it merged. Thanks.

**Pull request merged**